### ISSUE TITLE
PyChess now installable via KCP for KaOS

### ISSUE THREAD

**Issue by gbtami** [pychess/pychess#1718]
Hi all,

Latest PyChess (0.99.4) is now installable via KaOS's KCP (3rd party application), see here: https://kaos-community-packages.github.io/index2.html.

The actual package maintenance is on GitHub: https://github.com/KaOS-Community-Packages/pychess

KaOS is based on Arch Linux with focus on KDE / Qt only. GTK, Python are part of core system.

I would like that PyChess can be added on your list of available distributions, now also for KaOS (https://kaosx.us/).

PyChess on KaOS uses Python3 (v3.6.x).

I've managed to add all dependency packages in KCP that are not part of KaOS's main packages, in order for PyChess to run on KaOS.

There is one error that I'm not able to identify. When PyChess is run, it shows the following message on the console (when starting PyChess from shell):

Downloading https://api.github.com/repos/pychess/pychess/releases/latest ...
URL Error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:847) https://api.github.com/repos/pychess/pychess/releases/latest

If someone can guide me to the necessary steps to rectify this? KaOS supports various root certificates from https://certifi.io/en/latest/ via our python3-certifi core package. This is also installed, however the error is still reported by PyChess.. I guess it is another problem...

I'm not a Python developer, so I need carefull guidance, so this can be fixed on KaOS, having a 100% operational PyChess.

Best regards
Gunther

**Comment by gbtami** [comment]
I have the strong feeling that this is KaOS's python3 packaging problem, because this issue was never raised on any other distribution. At least I'v never got any complaint to date. All I can suggest is that you should consult your distribution core devs about this.

I'v looked into https://github.com/KaOS-Community-Packages/pychess/blob/master/PKGBUILD
I suggest to add stockfish to depends (PyChess learning modules will not work without that!) and remove gnuchess from optdepends the same time. Thx!

**Comment by gbtami** [comment]
What instruction should I add to http://pychess.org/download/ ?
Will `kcp -u && kcp -i pychess` download and install pychess on KaOS?

**Comment by bits4fun** [comment]
Hi @gbtami, to install PyChess using KCP, do (also stated on the github project):

kcp -i python3-pexpect
kcp -i python3-ptyprocess
kcp -i python3-psutil 
kcp -i python3-sqlalchemy 
kcp -i gtksourceview3 
kcp -i adwaita-icon-theme
kcp -i stockfish
and finally
kcp -i pychess

I will comment this with KaOS core team, re. SSL.

**Comment by bits4fun** [comment]
Stockfish is not a core package available in KaOS, however, I can create a new KCP package and let KCP pychess package depend on it. I didnt know that Stockfish was necesary for learning modules.

**Comment by gbtami** [comment]
Is `kcp` something like `dpkg` for Debian/Ubuntu ? I mean is it a low level package manager stuff which can't auto install dependencies? If this true, is there any higher level tool similar to `apt`(-get) with which can auto install dependencies listed in package meta data?

Dependencies for various distribution listed in https://github.com/pychess/pychess/blob/master/INSTALL

**Comment by bits4fun** [comment]
Yes, 

> Is `kcp` something like `dpkg` for Debian/Ubuntu ? I mean is it a low level package manager stuff which can't auto install dependencies? If this true, is there any higher level tool similar to `apt`(-get) with which can auto install dependencies listed in package meta data?
> 
> Dependencies for various distribution listed in https://github.com/pychess/pychess/blob/master/INSTALL

Yes, KCP is, I quote from the web site "KaOS users maintained set of files to easily build extra packages.".

It does not auto install dependencies of KCP packages, however, it installs packages referenced in KaOS repositories. That is why it is necessary to manually do "kcp" for packages created outside KaOS (unfortunately). There is only kcp available to install 3rd party packages for KaOS.

**Comment by bits4fun** [comment]
I have created new "stockfish" KCP (https://github.com/KaOS-Community-Packages/stockfish) and updated [pychess](https://github.com/KaOS-Community-Packages/pychess) KCP package to depend on it. I left GnuChess as optional, since it still is a usable engine for playing against it; for those users who wish to do so.

I have tested pychess, and yes, it picks up the engine definition, when the application starts up.
